File: Data Structures/Array/findPivotIndex.go
package main

func pivotIndex(arr []int) int {
	s := 0

	for _, v := range arr {
		s += v
	}

	ans, l, r := -1, 0, s

	for i, v := range arr {
		r -= v
		if l == r {
			ans = i
			break
		}
		l += v
	}
	return ans
}

File: Techniques/Prefix Sum/findPivotIndex.cpp
#include <vector>
// using namespace std;

class Solution {
    public:
        int pivotIndex(std::vector<int>& arr) {
            int left = 0, n = arr.size();
            int right = 0;

            for (int i = 1; i < n; i++) right += arr[i];

            int i = 0;

            while (left != right && i < n) {
                left += arr[i];
                right -= arr[i + 1];
                i++;
            }
            return left == right ? i : -1;
        }
};